src/parameterizations/lateral/MOM_self_attr_load.F90

```diff
@@ -41,47 +52,54 @@ module MOM_self_attr_load
 
 contains
 
-!> This subroutine calculates seawater self-attraction and loading based on sea surface height. This should
-!! be changed into bottom pressure anomaly in the future. Note that the SAL calculation applies to all motions
-!! across the spectrum. Tidal-specific methods that assume periodicity, i.e. iterative and read-in SAL, are
-!! stored in MOM_tidal_forcing module.
+!> This subroutine calculates seawater self-attraction and loading based on either sea surface height (SSH) or bottom
+!! pressure anomaly.  Note that the SAL calculation applies to all motions across the spectrum. Tidal-specific methods
+!! that assume periodicity, i.e. iterative and read-in SAL, are stored in MOM_tidal_forcing module.
+!! The input field can be either SSH [Z ~> m] or total bottom pressure [R L2 T-2 ~> Pa].  If total bottom pressure is
+!! used, bottom pressure anomaly is first calculated by subtracting a reference bottom pressure from an input file.
+!! The output field is expressed as geopotential height anomaly, and therefore has the unit of [Z ~> m].
 subroutine calc_SAL(eta, eta_sal, G, CS, tmp_scale)
-  type(ocean_grid_type), intent(in)  :: G  !< The ocean's grid structure.
+  type(ocean_grid_type), intent(in) :: G  !< The ocean's grid structure.
   real, dimension(SZI_(G),SZJ_(G)), intent(in)  :: eta     !< The sea surface height anomaly from
-                                                           !! a time-mean geoid [Z ~> m].
-  real, dimension(SZI_(G),SZJ_(G)), intent(out) :: eta_sal !< The sea surface height anomaly from
-                                                           !! self-attraction and loading [Z ~> m].
+              !! a time-mean geoid or total bottom pressure [Z ~> m] or [R L2 T-2 ~> Pa].
+  real, dimension(SZI_(G),SZJ_(G)), intent(out) :: eta_sal !< The geopotential height anomaly from
+              !! self-attraction and loading [Z ~> m].
   type(SAL_CS), intent(inout) :: CS !< The control structure returned by a previous call to SAL_init.
-  real,                   optional, intent(in)  :: tmp_scale !< A rescaling factor to temporarily convert eta
-                                                            !! to MKS units in reproducing sumes [m Z-1 ~> 1]
+  real, optional, intent(in)  :: tmp_scale !< A rescaling factor to temporarily convert eta
+              !! to MKS units in reproducing sumes [m Z-1 ~> 1]
 
   ! Local variables
+  real, dimension(SZI_(G),SZJ_(G)) :: bpa ! SSH or bottom pressure anomaly [Z ~> m or R L2 T-2 ~> Pa]
   integer :: n, m, l
```

Comment:
Please revise this comment to read:
`! SSH or bottom pressure anomaly [Z ~> m] or [R L2 T-2 ~> Pa]`.